plugins/org.jkiss.dbeaver.ext.hana.ui/src/org/jkiss/dbeaver/ext/hana/ui/views/HANAConnectionPage.java

```diff
@@ -290,21 +290,34 @@ private void editionUpdated() {
         }
         UIUtils.fixReadonlyTextBackground(databaseText);
 
-        toggleControlVisibility(instanceLabel);
-        toggleControlVisibility(instanceText);
-        toggleControlVisibility(databaseLabel);
-        toggleControlVisibility(databaseText);
+        boolean visible = edition != HANAEdition.GENERIC && edition != HANAEdition.CLOUD;
+        toggleControlVisibility(instanceLabel, visible);
+        toggleControlVisibility(instanceText, visible);
+        toggleControlVisibility(databaseLabel, visible);
+        toggleControlVisibility(databaseText, visible);
         ((Composite)getControl()).layout(true, true);
     }
 
-    private void toggleControlVisibility(Control control) {
-        control.setVisible(edition != HANAEdition.GENERIC);
+    private void toggleControlVisibility(Control control, boolean visible) {
+        control.setVisible(visible);
         Object layoutData = control.getLayoutData();
         if (layoutData instanceof GridData) {
-            ((GridData) layoutData).exclude = (edition == HANAEdition.GENERIC);
+            ((GridData) layoutData).exclude = !visible;
         }
     }
 
+    private void hostUpdated() {
+        if (CommonUtils.isEmpty(hostText.getText())) {
+            return;
+        }
+        String host = hostText.getText().trim();
+        if (edition == HANAEdition.GENERIC && CommonUtils.isEmpty(portText.getText()) && host.endsWith(":443")) {
```

Comment:
Let's move ":443" into constants (HTTPS_PORT_SUFFIX). 
So condition will be host.length() - HTTPS_PORT_SUFFIX.length()

Reply:
done